### PR TITLE
in the ApkDecoder use apkFile and unknownFiles from ApkInfo

### PR DESCRIPTION
**Pull request by sv99**
ApkDecoder and ResourceDecoder - use apkFile from ApkInfo

### CODE REVIEW

**Comment by iBotPeaches** [comment]
Adding @IgorEisberg or you to the refactors so we can all peek each other's.

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java [hunk 1]
```diff
@@ -295,7 +299,9 @@ public void write(YamlWriter writer) {
         writer.writeBool("resourcesAreCompressed", resourcesAreCompressed);
         writer.writeBool("sharedLibrary", sharedLibrary);
         writer.writeBool("sparseResources", sparseResources);
-        writer.writeStringMap("unknownFiles", unknownFiles);
+        if (unknownFiles.size() > 0) {
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java:302]: Shouldn't we always write this out even if empty? I know some tools parse the apktool.yml

Comment (sv99) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/apk/ApkInfo.java:302]: yes this is null and updated in the ApkDecoder.decode but after replace ResUnknownFiles - this field initialized when ApkInfo created. And for store behavior need test empty lines.

**Review by iBotPeaches — APPROVED** [review]
Tested it out. Looked for 2 things.

* If the move for unknown files still worked - it did
* If the file was properly closed after being set to an intermediate variable - it did

Somehow we are leaking a file close handler, but looks unrelated to this. I'll dig into that later.

**Comment by iBotPeaches** [comment]
thanks!
